### ISSUE TITLE
Adding new type of vehicle profile: ski

### ISSUE THREAD

**Issue by lains** [navit-gps/navit#759]
I was recently struggling with a map of a resort to find the quickest way to get home before ski-lifts close.
I have seen that openstreetmap has all info about ski-lifts and ski slopes (including difficulty).
Maybe this could be of interest to others, and navit is probably well positioned to implement this type of features.
Please react with +1 if you're interested in this feature.

**Comment by mvglasow** [comment]
Absolutely—in fact I have been thinking about this before, just never got around to working on it. There may be a few points to consider:

* **Routability:** As far as I remember, our routing engine is currently prepared to deal only with a specific range of item types, and I am not sure whether piste types fall within that range. If not, that would require some code changes. It might make sense to introduce `is_way_routable()` as a function or macro, and rely on that throughout the code—would make future changes a lot easier. (Like, if we extend the same to Nordic pistes.)
* **Access flags:** we don’t have any for skiers. Maybe we don’t even need them, as the vehicle profile governs which item types are to be used for routing, and access for skiers is clear from the fact that the way is tagged as a piste.
* **Vehicle profiles:** It may make sense to have multiple profiles for multiple skill levels (expressed as the highest level of difficulty to be used).
* **Multimode routing:** Skiers ski down pistes, take lifts and may need to walk short distances to when boarding or exiting certain types of lifts.
  * Routable lifts: we need an inventory of all lift types found in a ski resort, including more exotic ones (such as funiculars). 
  * Getting from a piste to a cable car may involve walking a short distance, using a way tagged as a footpath. We need to include these in the routing as well (else skiers will never be guided to that cable car in question) but make them sufficiently expensive (else skiers will be told to take off their skis and walk when there is a piste). Maybe start off with a copy of the pedestrian profile, assuming half the regular pedestrian speed for anything walkable (remember users are wearing ski boots and carrying a pair of skis). It might also make sense to introduce a penalty for taking off and putting on skis.
  * Taking a lift downhill: for some (e.g. drag lifts) it is not an option, for others (e.g. many chair lifts) it varies from lift to lift, and where possible, it is usually not the preferred option. We may need to penalize lifts taken downhill (not sure of the routing engine supports this, though).
* **Maneuvers:** We may need to introduce multiple new maneuver types. Taking a lift is much like taking a ferry. Then again, we don’t have any maneuver types for ferries either, so we might skip that in the first step and live with announcements such as “turn right into Télésiège de Rosaël”. If we do, new maneuvers would include “take the lift”, “take off skis” and “put on skis”.
* **Closures:** Lift and piste closures are much more frequent than for most other forms of transportation (except maybe for ferries), especially around the beginning or end of the season. Being told to go down a closed piste may be dangerous (say, if the closure is due to avalanche risk); in any case being told to go down a piste that is closed or a take a lift that is not in service defeats the purpose. OTOH, I am not really aware of a reliable source of information for these closures that Navit could use. Internally Navit would be able to do it, now that we have a working traffic module; we just need to get that information in there (and make sure the traffic module knows about pistes and skiers).
* **Poor mapping:** Until now, I frequently encounter instances in which pistes are not connected to lifts. That will break routing, and IMHO we should not try to fix this. Hopefully routing for skiers will raise some awareness here.

About the closure point—I have long (even before I learned about Navit) wanted a way to reject a maneuver (i.e. telling Navit “don’t do that right turn”), which could be implemented by adding a traffic distortion for the way immediately after the maneuver, marking it as closed. Maybe we could use that approach for lift and piste closures: when told do take a piste or lift that is closed, reject the maneuver, marking that lift or stretch of piste as closed. (Long-term goal would be to be able to share such information with others.) This should be accessible from the main navigation screen, ideally also from the route description (look at the route in advance and report items as closed).

Fun challenge: some roads are pistes in winter, which is expressed in OSM by double-tagging the same way. If the traffic module has a closure report for the road (which is likely), we need to ensure it does not affect skiers.

Another point about closures: sometimes lifts are temporarily closed to skiers but open to pedestrians. Maybe we do need access flags for skiers after all. Or maybe this is not a real issue, as closure of a lift to skiers usually also implies closure of pistes leading to and from it.

**Comment by hoehnp** [comment]
In this scope, I wonder if we should also consider enhancing our pedestrian profile to include public transport. In my opinion this seems pretty similar to sking.

**Comment by mvglasow** [comment]
That has a fundamental distinction from lifts, though: most lifts allow continuous boarding (escalator-style); at the most, aerial tramways or funiculars allow boarding only every few minutes (equal to the duration of a one-way trip, think of a longer-running elevator). Therefore, timetables are not a concern here; at most, operating hours are.

That is different with public transportation: during off-peak hours, you may have to wait some 20 minutes or more for the next train to arrive. If it takes less than that to walk the same distance, the arrival time of the next train is the main deciding factor for your choice of route. As far as I know this is the main reason why this has been regarded out of scope up until now.

Technically, public transportation is more akin to ferry routing—which we do support, albeit poorly (“go straight”—\*splash\*).

My suggestion would be to keep public transport routing separate, but keep it in mind as a future use case when we implement new logic that might be used by both.

**Comment by hoehnp** [comment]
Even escalator-style lifts can have non-deterministic delays due to queues. ;-) But in general I agree. My point was rather as you mentioned last, that one should keep it in mind, if new use-cases could also cover this one.
